### PR TITLE
return MOI.TIME_LIMIT when the solver times out

### PR DESCRIPTION
**Pull request by zengjian-hu-rai**
Should we return MOI.TIME_LIMIT instead of OTHER_ERROR when the solver hits the preset time_limit? 

Understand it's tricky to know whether the solver truly times out. Looks like the solver (chuffed) returns UNKNOWN when it hits the time limit.

### CODE REVIEW

**Review by odow — APPROVED** [review]
As a hack, this is okay. Does mini zinc not have better status reporting options?

**Comment by odow** [comment]
Looks like there isn't
```
--unsat-msg (--unsatisfiable-msg), --unbounded-msg, --unsatorunbnd-msg,
        --unknown-msg, --error-msg, --search-complete-msg <msg>
    Specify solution status messages. The defaults:
    "=====UNSATISFIABLE=====", "=====UNSATorUNBOUNDED=====", "=====UNBOUNDED=====",
    "=====UNKNOWN=====", "=====ERROR=====", "==========", respectively.
```

**Comment by chriscoey** [comment]
AFAICT these are the minizinc statuses https://minizinc-python.readthedocs.io/en/latest/api.html#minizinc.result.Status and it does not really have the level of granularity on termination statuses that MOI does

File: src/optimize.jl [hunk 1]
```diff
@@ -236,6 +236,10 @@ function MOI.get(model::Optimizer, ::MOI.TerminationStatus)
         else
             return MOI.OPTIMAL
         end
+    elseif model.solver_status == "UNKNOWN" &&
+           model.time_limit_sec !== nothing &&
+           abs(model.solve_time_sec - model.time_limit_sec) < 1
```
Comment (odow) [inline src/optimize.jl:241]:
```suggestion
           model.solve_time_sec > model.time_limit_sec + 0.1
```

Comment (odow) [reply src/optimize.jl:241]: Now I take another look, why the `abs`?

Comment (zengjian-hu-rai) [reply src/optimize.jl:241]: we can probably get rid of the abs, but not sure if 0.1s is enough. I 've seen solve_time_sec = 1.12s for time_limit_sec = 1s.

Comment (zengjian-hu-rai) [reply src/optimize.jl:241]: yes 1s seems too much - agree this is hacky. I 'm open to other ideas

Comment (odow) [reply src/optimize.jl:241]: I changed to just if the solve time exceeds the limit.

**Comment by odow** [comment]
Happy with this? Any changes?

**Comment by zengjian-hu-rai** [comment]
> Happy with this? Any changes?

yes!

**Pull request merged**